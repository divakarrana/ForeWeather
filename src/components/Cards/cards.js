import React from 'react'
import styles from './cards.module.css';

const cards = (props) => {

    //Waiting for the API response
    let allCards = (<div className={styles.loading}>Gathering weather data... <i className="fas fa-cog"></i></div>);

    //Successful API response results
    if(props.weatherData){
        allCards = props.weatherData.map( dayWeather => {

            //Formatting date 
            dayWeather.dt_txt =  new Date(dayWeather.dt_txt).toDateString();

            //Preparing JSX of cards with data
            return(
                <div className={styles.card} key={dayWeather.dt_txt} data-testid="card">
                    <div className={styles.cardHead}>
                        <div className={styles.cardHeadTemp}>{dayWeather.main.temp}&#176;</div>
                        <div className={styles.cardHeadDate}>{dayWeather.dt_txt}</div>
                        <div className={styles.cardHeadHumidity}><i className="fas fa-tint"></i>{dayWeather.main.humidity}% Humid</div>
                    </div>
                    <div className={styles.cardContent}>
                        <div className={styles.weatherStats}>
                            <i className="fas fa-cloud-sun"></i>Weather:&nbsp;{dayWeather.weather.map(el => el.main).join(",")}
                        </div>
                        <div className={styles.windStats}>
                            <i className="fas fa-wind"></i>Wind:&nbsp;{dayWeather.wind.speed}km/h
                        </div>
                        <div className={styles.pop}>
                            <i className="fas fa-cloud-rain"></i>Precipitation:&nbsp;{dayWeather.pop}%
                        </div>
                        <div className={styles.pop}>
                            <i className="fas fa-tachometer-alt"></i>Pressure:&nbsp;{dayWeather.main.pressure}hPa
                        </div>
                    </div>
                </div>
                );
        });
    }

    //Rendering Cards
    return (
        <div className={styles.container}>
            {allCards}
        </div>
    )
}

export default cards;
