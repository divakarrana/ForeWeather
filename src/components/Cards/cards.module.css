.container {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  align-items: center;
  margin-top: 4rem;
}

/* Loader */
.loading {
  color: var(--color-grey-light-1);
  font-size: 2.2rem;
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.loading i {
  animation: rotate 1s infinite;
}
/* OVERALL CARD CSS */
.card {
  color: var(--color-grey-light-1);
  background-image: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0.3),
      rgba(0, 0, 0, 0.3)
    ),
    url(../../assets/img/weather-night-min.jpg);
  background-size: cover;
  height: 30rem;
  width: 17rem;
  border-radius: 1rem;
  display: flex;
  flex-direction: column;
  margin: 3rem 2rem;
  box-shadow: var(--shadow-dark);
  overflow: hidden;
  transition: all 0.2s;
}

.card:hover {
  transform: translateY(-0.8rem);
  border: 2px solid var(--color-grey-light-1);
}

.cardHead {
  height: 60%;
  width: 100%;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.cardHeadTemp {
  font-size: 2.5rem;
  font-weight: 300;
}

.cardHeadDate {
  margin-bottom: 2.5rem;
}

.cardHeadDate,
.cardHeadHumidity {
  font-size: 1.1rem;
  font-weight: 600;
}

.cardContent {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: flex-start;
  height: 40%;
  width: 100%;
  margin-left: 1.2rem;
  font-size: 1.2rem;
}

.cardHeadHumidity i {
  margin-right: 1rem;
}

.cardContent i {
  margin-right: 1rem;
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(180deg);
  }
}
