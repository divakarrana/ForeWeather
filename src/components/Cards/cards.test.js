import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Cards from './cards';

describe('<Cards />', () => {

    afterEach(cleanup);

    it("should render without crashing <Cards />", () => {
        const div = document.createElement('div');
        ReactDOM.render(<Cards></Cards>, div);
    });

    it("should renders <Cards /> correctly", () => {
        const {getByTestId} = render(<Cards weatherData={[
            {
            "dt": 1605430800, 
            "main": {"temp": 31.37,"feels_like": 32.42,"temp_min": 30.17,"temp_max": 31.37,"pressure": 1011,"sea_level": 1011,"grnd_level": 1008,"humidity": 53,"temp_kf": 1.2},
            "weather": [{"id": 802,"main": "Clouds","description": "scattered clouds","icon": "03d"}],
            "clouds": {"all": 36},
            "wind": {"speed": 4.21,"deg": 313},
            "visibility": 10000,
            "pop": 0,
            "sys": {"pod": "d"},
            "dt_txt": "2020-11-15 09:00:00"
            }]}></Cards>);

            expect(getByTestId('card')).toBeInTheDocument();
    });

    it("should match snapshot <Cards />", () => {
        const tree = renderer.create(<Cards weatherData={[
            {
            "dt": 1605430800, 
            "main": {"temp": 31.37,"feels_like": 32.42,"temp_min": 30.17,"temp_max": 31.37,"pressure": 1011,"sea_level": 1011,"grnd_level": 1008,"humidity": 53,"temp_kf": 1.2},
            "weather": [{"id": 802,"main": "Clouds","description": "scattered clouds","icon": "03d"}],
            "clouds": {"all": 36},
            "wind": {"speed": 4.21,"deg": 313},
            "visibility": 10000,
            "pop": 0,
            "sys": {"pod": "d"},
            "dt_txt": "2020-11-15 09:00:00"
            }]}></Cards>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});