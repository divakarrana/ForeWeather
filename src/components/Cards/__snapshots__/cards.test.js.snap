// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> should match snapshot <Cards /> 1`] = `
<div
  className="container"
>
  <div
    className="card"
    data-testid="card"
  >
    <div
      className="cardHead"
    >
      <div
        className="cardHeadTemp"
      >
        31.37
        °
      </div>
      <div
        className="cardHeadDate"
      >
        Sun Nov 15 2020
      </div>
      <div
        className="cardHeadHumidity"
      >
        <i
          className="fas fa-tint"
        />
        53
        % Humid
      </div>
    </div>
    <div
      className="cardContent"
    >
      <div
        className="weatherStats"
      >
        <i
          className="fas fa-cloud-sun"
        />
        Weather: 
        Clouds
      </div>
      <div
        className="windStats"
      >
        <i
          className="fas fa-wind"
        />
        Wind: 
        4.21
        km/h
      </div>
      <div
        className="pop"
      >
        <i
          className="fas fa-cloud-rain"
        />
        Precipitation: 
        0
        %
      </div>
      <div
        className="pop"
      >
        <i
          className="fas fa-tachometer-alt"
        />
        Pressure: 
        1011
        hPa
      </div>
    </div>
  </div>
</div>
`;
